### PR TITLE
chore: fix docker push script

### PR DESCRIPTION
**Pull request by feng-tao**
### Summary of Changes

fix https://github.com/amundsen-io/amundsen/issues/658

### Tests

_What tests did you add or modify and why? If no tests were added or modified, explain why. Remove this line_

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes `make test`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,10 +27,21 @@ image:
...
 push-image:
-	docker push ${IMAGE}:${VERSION}
-	docker push ${IMAGE}:latest
+	docker push ${IMAGE}:latest .
+
+.PHONY: image-version
+image-version:
+	docker build -f public.Dockerfile -t ${IMAGE}:${VERSION} .
 
 .PHONY: oidc-image
```
Comment (feng-tao) [inline Makefile:46]: do we still need this image given we remove oidc in metadata? cc @verdan

File: .github/workflows/pypipublish.yml [hunk 2]
```diff
@@ -35,6 +35,9 @@ jobs:
         uses: actions/checkout@v1
       - name: Login with docker
         run: docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
-      - name: Publish to Registry
+      - name: Publish to Registry for latest
         if: success()
         run: make build-push-image
+      - name: Publish to Registry for version
+        if: startsWith(github.event.ref, 'refs/tags')
```
Comment (pPanda-beta) [inline .github/workflows/pypipublish.yml:42]: makes sense

**Review by pPanda-beta — COMMENTED** [review]
.

File: Makefile (hunk 1 above)
Comment (verdan) [reply Makefile:46]: Yes we still need this. The only thing we removed from metadata was extra OIDC config class. We still need to set `flaskoidc` package to enable oidc in metadata. and this image will be doing that.

**Review by verdan — APPROVED** [review]

**Pull request merged**